src/Predavanje02/LoginHelper.kt

```diff
@@ -0,0 +1,14 @@
+package Predavanje02
+
+import Predavanje02.Employee.Companion.employees
+
+class LoginHelper {
+    fun performLogin(username: String, onFailure: () -> Unit, onLogin: (Employee) -> Unit) {
+        val employee = employees.find { it.name == username }
+        if (employee != null) {
+            onLogin(employee)
+        } else {
+            onFailure()
+        }
+    }
+}
```

Comment:
Ovdje sam napravio primjer kako iskoristiti higher-order funkcije. Samo da dobiješ ideju. 

Ne moraš nužno vraćati zaposlenika kod logina. Možeš ga isto tako spremiti u neki singleton gdje ćeš mu imati pristup od svuda. Samo pazi u tom slučaju da ne može bilo tko pozvati .logout() ili odraditi nešto što ne bi smio moći.